From 1dab8848cdbb849908ded6fbb8a716bb11b6bce3 Mon Sep 17 00:00:00 2001
Message-Id: <1dab8848cdbb849908ded6fbb8a716bb11b6bce3.1424979244.git.feitong.yi@intel.com>
In-Reply-To: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
References: <d8466aca83b11ab8b0870e775536306782d73e9f.1424979244.git.feitong.yi@intel.com>
From: Gajanan Bhat <gajanan.bhat@intel.com>
Date: Fri, 30 Jan 2015 09:32:00 +0530
Subject: [PATCH 04/28] MUST_REBASE [VPG]: drivers/video/adf: Support ADF
 simple buffer alloc

When ADF adf_simple_buffer_alloc is called internally call
i915_adf_simple_buffer_alloc to create GEM buffer.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed.

Issue: GMINL-5535
Change-Id: I3b3bcde5d31e9dacbfb9a0bb5c417a82705c5e3d
Signed-off-by: Gajanan Bhat <gajanan.bhat@intel.com>
---
 drivers/video/adf/intel/intel_adf_interface.c |   11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/video/adf/intel/intel_adf_interface.c b/drivers/video/adf/intel/intel_adf_interface.c
index 198f532..c03126b 100644
--- a/drivers/video/adf/intel/intel_adf_interface.c
+++ b/drivers/video/adf/intel/intel_adf_interface.c
@@ -16,6 +16,9 @@
 #include <video/adf_client.h>
 #include "intel_adf.h"
 
+extern int i915_adf_simple_buffer_alloc(u16 w, u16 h, u8 bpp,
+			struct dma_buf **dma_buf, u32 *offset, u32 *pitch);
+
 /* Custom IOCTL */
 static long intel_interface_obj_ioctl(struct adf_obj *obj,
 	unsigned int cmd, unsigned long arg)
@@ -160,13 +163,7 @@ static int intel_interface_alloc_simple_buffer(struct adf_interface *intf,
 	dev_info(dev->base.dev, "%s: size %d\n", __func__, size);
 
 #ifdef CONFIG_ADF_INTEL_VLV
-	/*
-	 * For VLV we use GEM memory manager. Will be enabled when
-	 * we fix the FB driver. As of now memory is preallocate in
-	 * user space via DRM PRIME interface which gives DMA Buf
-	 * handles over GEM
-	 */
-	return -EOPNOTSUPP;
+	return i915_adf_simple_buffer_alloc(w, h, bpp, dma_buf, offset, pitch);
 #else
 	/*allocate buffer*/
 	return intel_adf_mm_alloc_buf(mm, size, dma_buf);
-- 
1.7.9.5

